### PR TITLE
Handle 422 when we update pull request branch

### PR DESCRIPTION
**Pull request by sileht**

### CODE REVIEW

File: mergify_engine/branch_updater.py [hunk 1]
```diff
@@ -168,7 +168,12 @@ def update_with_api(pull):
             input={"expected_head_sha": pull.g_pull.head.sha},
             headers={"Accept": "application/vnd.github.lydian-preview+json"})
     except github.GithubException as e:
-        if e.status < 500:
+        if e.status == 422:
+            LOG.debug("update branch done somewhere in meantime",
```
Comment (jd) [inline mergify_engine/branch_updater.py]: I'd go for "branch already updated" or "already up-to-date"

Comment (sileht) [reply mergify_engine/branch_updater.py]: Well maybe not, it can also be some-else that push more commits in the meantime

Comment (jd) [reply mergify_engine/branch_updater.py]: Ah ok I see what you mean. So maybe "branch updated in the meantime" would be clearer.

Comment (sileht) [reply mergify_engine/branch_updater.py]: done

**Pull request merged**